src/configuration/ConfigurationService.java

```diff
@@ -0,0 +1,89 @@
+package configuration;
+
+import com.esotericsoftware.yamlbeans.YamlReader;
+import com.esotericsoftware.yamlbeans.YamlWriter;
+
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Ali Bozna
+ */
+public class ConfigurationService {
+
+    public ConfigurationService() {
+    }
+
+    /**
+     * Deserializes the @see CrawlerConfig from the user path.
+     *
+     * @return Returns the @see CrawlerConfig.
+     */
+    public CrawlerConfig getConfig() {
```

Comment:
We need another function that we can call to simply get the already parsed Configuration.
This service should only parse the yaml file once and provide the CrawlerConfig with a getter.